src/Compilers/CSharp/Test/Semantic/Semantics/SemanticErrorTests.cs

```diff
@@ -19252,6 +19252,82 @@ static void Main()
                 VerifyDiagnostics(Diagnostic(ErrorCode.WRN_UnreachableCode, @"Console"));
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75605")]
+        public void CS0162WRN_UnreachableCode08_ThrowAfterScopedReturn()
+        {
+            var text = """
+                using System;
+                class Program
+                {
+                    static int Add(int a, int b)
+                    {
+                        {
+                            return a + b;
+                        }
+
+                        throw new InvalidOperationException();
+
+                        {
+                            throw new InvalidOperationException();
+                        }
+
+                        {
+                            const int x = 523;
+                            throw new InvalidOperationException();
+                        }
+
+                        {
+                            throw new InvalidOperationException();
+                        }
+                    }
+                }
+                """;
+            CreateCompilation(text).VerifyDiagnostics(
+                // (10,9): warning CS0162: Unreachable code detected
+                //         throw new InvalidOperationException();
```

Comment:
>throw new InvalidOperationException(); [](http://example.com/codeflow?start=27&length=38)

There are several lines like this in the code. Consider using a `#line` directive to make it clearer which line is this.